modules/core/src/store/actions/authenticate.ts

```diff
@@ -72,3 +73,13 @@ export const setDefaultTenant = (tenantName: string): SetDefaultTenantActionInte
     payload: tenantName,
     type: CommonAuthenticateActionTypes.SET_DEFAULT_TENANT
 });
+
+/**
+ * Redux action to set tenants.
+ *
+ * @return {SetTenantsActionInterface} An action of type `SET_TENANTS`.
+ */
+export const setTenants = (tenants: any[]): SetTenantsActionInterface => ({
```

Comment:
Let's try to use generics here. Send a followup PR.
```suggestion
export const setTenants = <T>(tenants: T[]): SetTenantsActionInterface => ({
```

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/3122/commits/2966e0cf97aaf23f33023716b2c77d314d3179c3